code/main/edge/edgeRequestWorker.brs

```diff
@@ -210,19 +312,49 @@ function _adb_EdgeRequestWorker(edgeResponseManager as object) as object
         _createEdgeRequestBody: function(eventData as object, ecid as string, meta as object) as object
             requestBody = {
                 "xdm": {
-                    "identityMap": m._getIdentityMap(ecid),
                     "implementationDetails": _adb_ImplementationDetails()
                 },
                 "events": []
             }
 
-            ''' Add eventData under events key as an array
+            if not _adb_isEmptyOrInvalidString(ecid)
+                ' Add ECID to the xdm.identityMap
+                requestBody.xdm.identityMap = m._getIdentityMap(ecid)
+            end if
+
+            ' Add eventData under events key as an array
             requestBody.events = [eventData]
 
             if not _adb_isEmptyOrInvalidMap(meta)
                 requestBody.meta = meta
             end if
 
+            _adb_logVerbose("EdgeRequestWorker::_createEdgeRequestBody() - Created Edge request body: (" + FormatJson(requestBody) + ").")
+            return requestBody
+        end function,
+
+        _createConsentRequestBody: function(consentData as object, ecid as string, meta as object) as object
+            requestBody = {
+                "query" : {
+                    "consent" : {
+                    "operation" : "update"
+                    }
+                },
+                "xdm": {
+                    "identityMap": m._getIdentityMap(ecid),
```

Comment:
Do you need the same logic here as done in `_createEdgeRequestBody` to check if `ecid` is invalid or empty? Will Konductor generate and return an ECID value for update consent requests?

Reply:
Yes, we will be removing ECID requirement later, I can revert this change for now, but our logic will not work since identity is blocked by consent and we might not have ECID. This can be a part of new task #188